tests/TestData/Providers/Reflection/ReflectionClassesTestDataProviders.php

```diff
@@ -0,0 +1,50 @@
+<?php
+declare(strict_types=1);
+
+namespace StubTests\TestData\Providers\Reflection;
+
+use Generator;
+use StubTests\Model\PHPClass;
+use StubTests\Model\PHPInterface;
+use StubTests\Model\StubProblemType;
+use StubTests\TestData\Providers\EntitiesFilter;
+use StubTests\TestData\Providers\ReflectionStubsSingleton;
+
+class ReflectionClassesTestDataProviders
+{
+    public static function allClassesProvider(): ?Generator
+    {
+        $allClassesAndInterfaces = ReflectionStubsSingleton::getReflectionStubs()->getClasses() +
+            ReflectionStubsSingleton::getReflectionStubs()->getInterfaces();
+        foreach (EntitiesFilter::getFiltered($allClassesAndInterfaces) as $class) {
+            //exclude classes from PHPReflectionParser
+            if (strncmp($class->name, 'PHP', 3) !== 0) {
+                yield "class {$class->name}" => [$class];
+            }
+        }
+    }
+
+    public static function classesWithInterfacesProvider(): ?Generator
+    {
+        foreach (EntitiesFilter::getFiltered(ReflectionStubsSingleton::getReflectionStubs()->getClasses(),
+            fn(PHPClass $class) => empty($class->interfaces), StubProblemType::WRONG_INTERFACE) as $class) {
```

Comment:
Shouldn't it be `!empty($class->interfaces)`?

Reply:
`getFiltered` excludes some entities and returns left ones so actually this closure defines entities that will be removed from the result list e.g. classes that don't have any interfaces + classes that have muted problem `StubProblemType::WRONG_INTERFACE`